ui/src/main/js/application/auth/SamlForm.js

```diff
@@ -0,0 +1,340 @@
+                                errorName={FieldModelUtilities.createFieldModelErrorKey(AUTHENTICATION_SAML_GLOBAL_FIELD_KEYS.metadataUrl)}
+                                errorValue={errors.fieldErrors[AUTHENTICATION_SAML_GLOBAL_FIELD_KEYS.metadataUrl]}
+                            />
+                            <LabeledField label={importBlackDuckSSOConfigLabel} description={importBlackDuckSSOConfigDescription}>
+                                <div className={classes.fillForm}>
+                                    <GeneralButton id="blackduck-sso-import-button" disabled={!samlEnabled} onClick={() => setShowBlackDuckSSOImportModal(true)}>Fill Form</GeneralButton>
+                                </div>
+                            </LabeledField>
+                            <BlackDuckSSOConfigImportModal
+                                label={importBlackDuckSSOConfigLabel}
+                                csrfToken={csrfToken}
+                                readOnly={readonly || !samlEnabled}
+                                show={showBlackDuckSSOImportModal}
+                                onHide={() => setShowBlackDuckSSOImportModal(false)}
+                                initialSSOFieldData={formData}
+                                updateSSOFieldData={(data) => setFormData(data)}
+                            />
+                        </>
+                    )
+                }
+
+                { formData.metadataMode === 'FILE' &&
+                    <UploadFileButtonField
+                        id={AUTHENTICATION_SAML_GLOBAL_FIELD_KEYS.metadataFilePath}
+                        name={AUTHENTICATION_SAML_GLOBAL_FIELD_KEYS.metadataFilePath}
+                        fieldKey={AUTHENTICATION_SAML_GLOBAL_FIELD_KEYS.metadataFilePath}
+                        label="Identity Provider Metadata File"
+                        description="The file to upload to the server containing the Metadata from the external Identity Provider."
+                        readOnly={!samlEnabled && !displayTest}
+                        permissions={{ read: fileRead, write: fileWrite, delete: fileDelete }}
+                        onChange={FieldModelUtilities.handleConcreteModelChange(formData, setFormData)}
+                        buttonLabel="Upload"
+                        endpoint="/alert/api/authentication/saml/upload/metadata"
+                        customEndpoint="/alert/api/authentication/saml/upload/metadata"
+                        csrfToken={csrfToken}
+                        capture=""
+                        multiple={false}
+                        accept={[
+                            'text/xml',
+                            'application/xml',
+                            '.xml'
+                        ]}
+                        currentConfig={formData}
+                        value={FieldModelUtilities.getFieldModelSingleValue(formData, AUTHENTICATION_SAML_GLOBAL_FIELD_KEYS.metadataFilePath)}
```

Comment:
For some of these getFieldModelSingleValue calls, can we just directly call formData.<attribute> like formData.metadataFilePath here?